### ISSUE TITLE
github acion panic(main thread): index out of bounds: index 681, len 680

### ISSUE THREAD

**Issue by linzili** [oven-sh/bun#12115]
### How can we reproduce the crash?

gitaction 
windows-latest os
Run bun install
  bun install
  shell: C:\Program Files\PowerShell\7\pwsh.EXE -command ". '{0}'"
  env:
    CARGO_HOME: C:\Users\runneradmin/.cargo
    CARGO_INCREMENTAL: 0
    CARGO_TERM_COLOR: always
[0.00ms] ".env.production", ".env"
bun install v1.1.16 (bf7b3[2](https://github.com/linzili/iot-hzwl-web/actions/runs/9642587432/job/26590755466#step:7:2)7f)
Resolving dependencies
Resolved, downloaded and extracted [56]
============================================================
Bun v1.1.16 (bf7b[3](https://github.com/linzili/iot-hzwl-web/actions/runs/9642587432/job/26590755466#step:7:3)27f) Windows x64
Args: "C:\Users\runneradmin\.bun\bin\bun.exe" "install"
Features: dotenv(2) extracted_packages(179) 
Elapsed: [4](https://github.com/linzili/iot-hzwl-web/actions/runs/9642587432/job/26590755466#step:7:4)8923ms | User: 36[5](https://github.com/linzili/iot-hzwl-web/actions/runs/9642587432/job/26590755466#step:7:5)6ms | Sys: 11953ms
RSS: 0.84GB | Peak: 1.10GB | Commit: 0.9[6](https://github.com/linzili/iot-hzwl-web/actions/runs/9642587432/job/26590755466#step:7:6)GB | Faults: 498985

panic(main thread): index out of bounds: index 6[8](https://github.com/linzili/iot-hzwl-web/actions/runs/9642587432/job/26590755466#step:7:9)1, len 680
oh no: Bun has crashed. This indicates a bug in Bun, not your code.

To send a redacted crash report to Bun's team,
please file a GitHub issue using the link below:

### Relevant log output

```shell
Run bun install
  bun install
  shell: C:\Program Files\PowerShell\7\pwsh.EXE -command ". '{0}'"
  env:
    CARGO_HOME: C:\Users\runneradmin/.cargo
    CARGO_INCREMENTAL: 0
    CARGO_TERM_COLOR: always
[0.00ms] ".env.production", ".env"
bun install v1.1.16 (bf7b3[2](https://github.com/linzili/iot-hzwl-web/actions/runs/9642587432/job/26590755466#step:7:2)7f)
Resolving dependencies
Resolved, downloaded and extracted [56]
============================================================
Bun v1.1.16 (bf7b[3](https://github.com/linzili/iot-hzwl-web/actions/runs/9642587432/job/26590755466#step:7:3)27f) Windows x64
Args: "C:\Users\runneradmin\.bun\bin\bun.exe" "install"
Features: dotenv(2) extracted_packages(179) 
Elapsed: [4](https://github.com/linzili/iot-hzwl-web/actions/runs/9642587432/job/26590755466#step:7:4)8923ms | User: 36[5](https://github.com/linzili/iot-hzwl-web/actions/runs/9642587432/job/26590755466#step:7:5)6ms | Sys: 11953ms
RSS: 0.84GB | Peak: 1.10GB | Commit: 0.9[6](https://github.com/linzili/iot-hzwl-web/actions/runs/9642587432/job/26590755466#step:7:6)GB | Faults: 498985

panic(main thread): index out of bounds: index 6[8](https://github.com/linzili/iot-hzwl-web/actions/runs/9642587432/job/26590755466#step:7:9)1, len 680
oh no: Bun has crashed. This indicates a bug in Bun, not your code.

To send a redacted crash report to Bun's team,
please file a GitHub issue using the link below:
```

### Stack Trace (bun.report)

Bun v1.1.16 ([`bf7b327`](<https://github.com/oven-sh/bun/tree/bf7b327f68568104a5e40c1944b7b255b02f0745>)) on windows x86_64 [InstallCommand]

**panic**: index out of bounds: index 681, len 680

- [`install.zig:4906`](<https://github.com/oven-sh/bun/blob/bf7b327f68568104a5e40c1944b7b255b02f0745/src/install/install.zig#L4906>): `enqueueDependencyWithMain`
- [`install.zig:5793`](<https://github.com/oven-sh/bun/blob/bf7b327f68568104a5e40c1944b7b255b02f0745/src/install/install.zig#L5793>): `processDependencyList`
- [`install.zig:12861`](<https://github.com/oven-sh/bun/blob/bf7b327f68568104a5e40c1944b7b255b02f0745/src/install/install.zig#L12861>): `isDone`
- [`install.zig:12899`](<https://github.com/oven-sh/bun/blob/bf7b327f68568104a5e40c1944b7b255b02f0745/src/install/install.zig#L12899>): `installPackages`
- [`install.zig:13637`](<https://github.com/oven-sh/bun/blob/bf7b327f68568104a5e40c1944b7b255b02f0745/src/install/install.zig#L13637>): `installWithManager`
- [`filter_run.zig:431`](<https://github.com/oven-sh/bun/blob/bf7b327f68568104a5e40c1944b7b255b02f0745/src/cli/filter_run.zig#L431>): `install`
- [`cli.zig:1494`](<https://github.com/oven-sh/bun/blob/bf7b327f68568104a5e40c1944b7b255b02f0745/src/cli.zig#L1494>): `start`
- [`cli.zig:62`](<https://github.com/oven-sh/bun/blob/bf7b327f68568104a5e40c1944b7b255b02f0745/src/cli.zig#L62>): `start`
- [`main.zig:51`](<https://github.com/oven-sh/bun/blob/bf7b327f68568104a5e40c1944b7b255b02f0745/src/main.zig#L51>): `main`
- [`exe_common.inl:78`](<https://github.com/oven-sh/bun/blob/bf7b327f68568104a5e40c1944b7b255b02f0745/src/src/startup/exe_common.inl#L78>): `invoke_main`

**Comment by dylan-conway** [comment]
Thanks for the bug report. Can you share the `package.json` and `bun.lockb` for this project?

**Comment by dylan-conway** [comment]
Fixed with #12485 in bun v1.1.20
